### PR TITLE
[WIP] Cleaning deprecated methods

### LINKED ISSUE #382: Remove all deprecated code
Remove all deprecated code introduced in powsybl-1.x.x

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
Fix #382

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/VariantManager.java [hunk 1]
```diff
@@ -6,8 +6,71 @@
  */
 package com.powsybl.iidm.network;
 
+import java.util.Collection;
+import java.util.List;
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-public interface VariantManager extends StateManager {
+public interface VariantManager {
```
Comment (yichen88) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/VariantManager.java]: Missing docs from StateManager

Comment (ChamseddineBhd) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/VariantManager.java]: Docs added

**Comment by ChamseddineBhd** [comment]
I need to check rte-core before merging.

**Comment by mathbagu** [comment]
The groovy bindings are missing (at least for getShunt)

**Comment by MioRtia** [comment]
You missed some deprecated methods (e.g. ```Network.getStateManager``` )

**Comment by ChamseddineBhd** [comment]
I didn't delete the last deprecated method "setDefaultConfig" in the PlatformConfig class because that creates some problems. I'll create a new branch to fix those problems.

**Pull request merged**